### PR TITLE
CI: Moves to zero-root indexing

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
As discussed CI moves to a zero-indexed scheme.

## Todos
* **Developer Interest**
  - [x] Zero-indexing scheme.
  - [x] Returned value for CAS was returning lowest root instead of average for SA computations. 
* **User-Facing for Release Notes**
  - [x] CI Roots are now zero index (the ground state is now 0 instead of 1).

## Status
- [x]  Ready to go

### CODE REVIEW

File: psi4/src/read_options.cc [hunk 1]
```diff
@@ -489,7 +489,7 @@ int read_options(const std::string &name, Options & options, bool suppress_print
     (the one-particle density matrices are written for all roots).
     Useful for a state-specific CASSCF or CI optimization on an
     excited state. -*/
-    options.add_int("FOLLOW_ROOT", 1);
+    options.add_int("FOLLOW_ROOT", 0);
```
Comment (loriab) [inline psi4/src/read_options.cc]: Also,
* https://github.com/psi4/psi4/blob/master/psi4/src/read_options.cc#L318
* https://github.com/psi4/psi4/blame/master/doc/sphinxman/source/glossary_psivariables.rst#L340


Can https://github.com/psi4/psi4/blob/master/psi4/src/read_options.cc#L2075 be changed cosmetically easily?

@fevangelista, https://github.com/psi4/psi4/blob/master/psi4/src/read_options.cc#L2283 is typically ground state, yes? Can it's external ordering be switched to 0?

File: psi4/src/psi4/detci/opdm.cc [hunk 2]
```diff
@@ -571,120 +570,6 @@ void CIWavefunction::opdm_block(struct stringwr **alplist, struct stringwr **bet
 
 }// End OPDM Block
 
-/*
-** Here we compute all properties for opdms in opdm_map_ from form_opdm
-*/
```
Comment (loriab) [inline psi4/src/psi4/detci/opdm.cc]: This whole section is now what's handled by `oeprop()` in driver?

File: psi4/driver/procedures/proc.py [hunk 3]
```diff
@@ -2594,7 +2594,7 @@ def run_detci_property(name, **kwargs):
         core.print_out("\n   ===> %s properties for all CI roots <=== \n\n" % name.upper())
         for root in states:
             oe.set_title("%s ROOT %d" % (name.upper(), root))
-            root = root - 1
+            # root = root - 1
```
Comment (loriab) [inline psi4/driver/procedures/proc.py]: Is there a comment in detci somewhere about external/internal root numbering? Are they equivalent now?

File: psi4/src/psi4/detci/opdm.cc (hunk 2 above)
Comment (dgasmith) [inline psi4/src/psi4/detci/opdm.cc]: Yup, no need for it C side.

File: psi4/src/read_options.cc (hunk 1 above)
Comment (loriab) [inline psi4/src/read_options.cc]: Ah, sorry, on third point. I read the description and thought it was plain SCF.

**Comment by dgasmith** [comment]
This should be ready to go and is rebased up to master.

**Pull request merged**